src/identityRestorer.ts

```diff
@@ -0,0 +1,41 @@
+import axios from 'axios';
+
+export interface IdentityRestorerInterface {
+  addressHasBeenUsed(address: string): Promise<boolean>;
+  addressesHaveBeenUsed(addresses: string[]): Promise<boolean[]>;
+}
+
+/**
+ * Implementation of IdentityRestorerInterface using Esplora endpoint.
+ */
+export default class EsploraIdentityRestorer
+  implements IdentityRestorerInterface {
+  static DEFAULT_ESPLORA_ENDPOINT: string = 'http://localhost:3001';
```

Comment:
I woudl optimized for mainnet. 

```suggestion
  static DEFAULT_ESPLORA_ENDPOINT: string = 'https://blockstream.com/liquid/api';
```